### PR TITLE
Add GptOssForSequenceClassification for GPT-OSS models

### PR DESCRIPTION
**Pull request by zyfedward**
# What does this PR do?

Add GptOssForSequenceClassification for GPT-OSS models following generic practice for #40050 

Testing:
- Tested with single-node (8 A100) and multi-node (16 A100) options, loss curve and AUC are all expected with no issue.
- All testes passed
- `make fixup` passed

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request), Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@ArthurZucker @Rocketknight1

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -890,8 +890,11 @@ def _load_state_dict_into_meta_model(
                     param_name = hf_quantizer.update_param_name(param_name)
                     module, param_type = get_module_from_name(model, param_name)
                     value = getattr(module, param_type)
-                    # special case for GptOssForCausalLM, we wait for the param to be leave the meta device before casting it to cpu
-                    if model.__class__.__name__ == "GptOssForCausalLM" and value.device.type == "meta":
+                    # special case for GptOssForCausalLM and GptOssForSequenceClassification, we wait for the param to be leave the meta device before casting it to cpu
+                    if (
+                        model.__class__.__name__ in ["GptOssForCausalLM", "GptOssForSequenceClassification"]
```
Comment (ArthurZucker) [inline src/transformers/modeling_utils.py]: we'd probably better test the model's config type

**Review by ArthurZucker — COMMENTED** [review]
Make sure tests are unrelated, otherwise LGTM

**Comment by zyfedward** [comment]
> Make sure tests are unrelated, otherwise LGTM

Confirmed unrelated, no error found in circleci but timeout: `Too long with no output (exceeded 10m0s): context deadline exceeded`.

**Review by ArthurZucker — APPROVED** [review]
Cool merging!

**Comment by chutzpah121** [comment]
I’m training with GptOssForSequenceClassification using the Hugging Face Trainer, and I’ve noticed that both the loss and grad norm are very large. I suspect the losses are not being averaged properly.
For example, when I train on 32 × H20 GPUs, the loss is about 4× larger than when I train on 8 × H20 GPUs.
Has anyone encountered a similar issue?
Environment:
transformers==4.56.2
accelerate==1.10.1
